### ISSUE TITLE
Incorrect Default Config in Platform-Specific Button Actions and Links for Pinpoint In-App Messages affecting V5 and V6

### ISSUE THREAD

**Issue by buddyeorl** [aws-amplify/amplify-js#12915]
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

Angular, React, React Native, Vue

### Amplify APIs

Push Notifications

### Amplify Version

v6

### Amplify Categories

notifications

### Backend

Amplify CLI

### Environment information

<details>

```
# Put output below this line


```

</details>

### Describe the bug

There's an issue in the extractContent function within In `notifications/inAppMessaging/providers/pinpoint/utils/helpers/utils.ts`, where platform-specific primary and secondary button action overrides in both web and React Native applications are not functioning as intended. Instead of applying the specified overrides for different platforms, the function always returns the default configuration. This behavior leads to incorrect handling of button actions and links, particularly when using deep links or specific navigation triggers in segmented user campaigns.

For instance, if a default configuration is set to a regular URL (e.g., https://myapp.com/settings) and overrides are intended for iOS and Android (e.g., com.myapp://settings), the iOS and Android apps mistakenly try to open the default URL (https://myapp.com/settings) instead of the overridden deep link. Conversely, if the default is set as a deep link and the override is intended for Web Config, browsers fail to recognize the scheme and issue a warning, attempting to open the deep link instead of the appropriate web URL.

### Expected behavior

Platform-Specific Overrides: If a Pinpoint in-app message includes overrides for certain platforms, such as iOS or Android, these overrides should take precedence over the default configuration when the message is accessed on those specific platforms. For example, if the default link for a button in the message is set to https://myapp.com/settings, but an override for Android is specified as com.myapp://settings, then on Android devices, the message should use the com.myapp://settings link.

Default Configuration as Fallback: In cases where a specific platform does not have an override, the message should gracefully revert to using the default configuration. This ensures that the message remains functional and relevant even on platforms without specific overrides.

### Reproduction steps

- Create a Pinpoint Message: Set a default URL for button actions/links in an AWS Pinpoint in-app message.
- Add Overrides: Include deep link overrides for platforms like Android and iOS (e.g., com.myapp://settings).
- Test on Mobile: Open the message in a mobile app and observe the button link behavior.
- Test on Web: Open the same message in a web browser and check the button link behavior.

### Code Snippet

```javascript
export const extractContent = ({
	InAppMessage: message,
}: PinpointInAppMessage): InAppMessageContent[] => {
	return (
		message?.Content?.map(content => {
			const {
				BackgroundColor,
				BodyConfig,
				HeaderConfig,
				ImageUrl,
				PrimaryBtn,
				SecondaryBtn,
			} = content;
			const defaultPrimaryButton = PrimaryBtn?.DefaultConfig;
			const defaultSecondaryButton = SecondaryBtn?.DefaultConfig;
			const extractedContent: InAppMessageContent = {};
			if (BackgroundColor) {
				extractedContent.container = {
					style: {
						backgroundColor: BackgroundColor,
					},
				};
			}
			if (HeaderConfig) {
				extractedContent.header = {
					// Default to empty string in rare cases we don't have a Header value
					content: HeaderConfig.Header ?? '',
					style: {
						color: HeaderConfig.TextColor,
						textAlign:
							HeaderConfig.Alignment?.toLowerCase() as InAppMessageTextAlign,
					},
				};
			}
			if (BodyConfig) {
				extractedContent.body = {
					// Default to empty string in rare cases we don't have a Body value
					content: BodyConfig.Body ?? '',
					style: {
						color: BodyConfig.TextColor,
						textAlign:
							BodyConfig.Alignment?.toLowerCase() as InAppMessageTextAlign,
					},
				};
			}
			if (ImageUrl) {
				extractedContent.image = {
					src: ImageUrl,
				};
			}
			if (defaultPrimaryButton) {
				extractedContent.primaryButton = {
					// Default to empty string in rare cases we don't have a Text value
					title: defaultPrimaryButton.Text ?? '',
					action: defaultPrimaryButton.ButtonAction as InAppMessageAction,
					url: defaultPrimaryButton.Link,
					style: {
						backgroundColor: defaultPrimaryButton.BackgroundColor,
						borderRadius: defaultPrimaryButton.BorderRadius,
						color: defaultPrimaryButton.TextColor,
					},
				};
			}
			if (defaultSecondaryButton) {
				extractedContent.secondaryButton = {
					// Default to empty string in rare cases we don't have a Text value
					title: defaultSecondaryButton.Text ?? '',
					action: defaultSecondaryButton.ButtonAction as InAppMessageAction,
					url: defaultSecondaryButton.Link,
					style: {
						backgroundColor: defaultSecondaryButton.BackgroundColor,
						borderRadius: defaultSecondaryButton.BorderRadius,
						color: defaultSecondaryButton.TextColor,
					},
				};
			}
			return extractedContent;
		}) ?? []
	);
};

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

**Comment by nadetastic** [comment]
Hi @buddyeorl thank you for raising this issue, identifying a fix and submitting your contribution on #12917!!

I'm discussing this with the team and will follow up with any additional info. In the meantime, let me know if you have any questions.

**Comment by cwomack** [comment]
@buddyeorl, the PR was merged and the fix should now be released! Please upgrade to the latest version (v6.0.29) and let us know if you can confirm it's resolved.

**Comment by cwomack** [comment]
If anyone experiences this further in `v6.0.29` or later, feel free to comment/respond back.  Closing this issue as resolved.
